test/test_integration.py

```diff
@@ -0,0 +1,114 @@
+from killbill.rest import ApiException
+
+
+
+class TestIntegration(unittest.TestCase):
+    """Account unit test stubs"""
+
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    def test_integration(self):
+
+        # Create tenant
+        # TODO WORKING WITH ISSUES*! [ValueError: Invalid value for `api_key`, must not be `None`]
+        tenant = killbill.api.TenantApi()
+        body = Tenant(api_key='bob_5', api_secret='lazar_5')
+        tenant.create_tenant(body, 'test')
+
+        # Get tenant
+        # TODO NOT WORKING! [ApiException: (401) Reason: Unauthorized] or [ValueError: Invalid value for `api_secret`, must not be `None`]
+        body = Tenant(api_key='bob')
+        tenant.get_tenant_by_api_key(body)
+
+        # Upload XML catalog/Fetch XML catalog
+        # TODO NOT WORKING! [ApiException: (404) Reason: Not Found]
+        catalog = killbill.api.CatalogApi()
+        xml_catalog = open("/home/jfrodriguez/Documents/KillBill-Docs/SpyCarBasic.xml", "r+").read()
+        catalog.upload_catalog_xml(xml_catalog, 'test', 'bob', 'lazar')
+
+        # Create account
+        # WORKING!
+        account = killbill.api.AccountApi()
+        body = Account(name='Johny', currency='USD', state='CA', country='USA')
+        account.create_account(body, 'test', 'bob', 'lazar')
+
+        # Get account
+        # TODO NOT WORKING! [AttributeError: 'module' object has no attribute 'Str']
+        account = account.get_account_by_key('test_external_key', 'bob', 'lazar')
+        account = account.get_account('4eba7df9-f023-4425-bb46-b1a2e6e84fea', 'bob', 'lazar')
+
+        # Add a payment method
+        # WORKING!
+        account = killbill.api.AccountApi()
+        account_id = 'b506e9bc-0afd-495c-a047-a1eada0599be'
+        body = PaymentMethod(plugin_name='__EXTERNAL_PAYMENT__', plugin_info=None)
+        account.create_payment_method(account_id, body, 'test', 'bob', 'lazar')
+
+        # Get a payment method
+        # TODO NOT WORKING! [AttributeError: 'module' object has no attribute 'List[PaymentMethod]']
+        account.get_payment_methods_for_account(account_id, 'bob', 'lazar')
+
+        # Tag account as AUTO_INVOICING_OFF
+        # TODO NOT WORKING! [Bad Request: POST http://localhost:8080/1.0/kb/accounts/b506e9bc-0afd-495c-a047-a1eada0599be/tags?tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=-&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=-&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=-&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=-&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=2]
+        account_id = 'b506e9bc-0afd-495c-a047-a1eada0599be'
+        account.create_account_tags(account_id, 'test', 'bob', 'lazar')
+
+        # Create a subscription against plan
+        # TODO NOT WORKING! [ValueError: Invalid value for `product_name`, must not be `None`]
```

Comment:
Only `plan_name` or all of (`product_name`,  `billing_period ` and `price_list`) should be specified, but not a combination.

Reply:
If I try:
```
body = Subscription(account_id='4eba7df9-f023-4425-bb46-b1a2e6e84fea',
                    product_name='Basic',
                    billing_period='MONTHLY',
                    price_list='DEFAULT') 
 ```
Python throws the following error:
`ValueError: Invalid value for plan_name, must not be None`

Otherwise if I try:
```
body = Subscription(account_id='4eba7df9-f023-4425-bb46-b1a2e6e84fea',
                    plan_name='sports-monthly')
```
Python throws the following error:
`ValueError: Invalid value for product_name, must not be None`